charts/external-dns/templates/deployment.yaml

```diff
@@ -1,3 +1,5 @@
+---
+{{- $providerName := include "external-dns.providerName" . }}
```

Comment:
```suggestion
{{- $providerName := include "external-dns.providerName" . }}
```

Reply:
Only if it starts with directives. Helm chart YAML files idiomatically don't start with a `---`; you can run `helm new` to prove this (on top of just checking the many charts out there). When merging templates Helm adds document separators so the only time a manual separator is required in a Helm template is when you're creating multiple documents from a single template such as by using `range`.